tango/step.py

```diff
@@ -354,18 +359,31 @@ def _run_with_work_dir(self, workspace: "Workspace", **kwargs) -> T:
             self.work_dir_for_run = Path(dir_for_cleanup.name)
 
         try:
-            if self.cache_results:
-                workspace.step_starting(self)
```

Comment:
We used to have a thing where workspaces never get bothered with uncacheable results. It's also OK if we say, workspaces must handle uncacheable results at all times. But we have to be consistent about it.